opencog/cython/PythonEval.cc

```diff
@@ -609,13 +622,83 @@ std::string PythonEval::apply_script(const std::string& script)
 
 void PythonEval::addSysPath(std::string path)
 {
-    PyList_Append(this->pySysPath, PyBytes_FromString(path.c_str()));
-    //    PyRun_SimpleString(("sys.path += ['" + path + "']").c_str());
+    PyObject* pyPathString = PyBytes_FromString(path.c_str());
+    PyList_Append(this->pySysPath, pyPathString);
+
+    // We must decrement because, unlike PyList_SetItem, PyList_Append does
+    // not "steal" the reference we pass to it. So this:
+    //
+    // PyList_Append(this->pySysPath, PyBytes_FromString(path.c_str()));
+    // 
+    // leaks memory. So we need to save the reference as above and
+    // decrement it, as below.
+    //
+    Py_DECREF(pyPathString);
 }
 
 
 const int ABSOLUTE_IMPORTS_ONLY = 0;
 
+void PythonEval::import_module( const boost::filesystem::path &file,
+                                PyObject* pyFromList)
+{
+    // The pyFromList parameter corresponds to what would appear in an
+    // import statement after the import:
+    //
+    // from <module> import <from list>
+    //
+    // When this list is empty, this corresponds to an import of the
+    // entire module as is done in the simple import statement:
+    //
+    // import <module>
+    //
+    string fileName, moduleName;
+
+    // Get the module name from the Python file name by removing the ".py"
+    fileName = file.filename().c_str();
+    moduleName = fileName.substr(0, fileName.length()-3);
```

Comment:
hmmm, just wondering, what happens if the module is a directory? In that case the file name would be `__init__`